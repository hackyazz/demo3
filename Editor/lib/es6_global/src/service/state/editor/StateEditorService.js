

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
