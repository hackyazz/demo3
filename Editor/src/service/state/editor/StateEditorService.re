open EditorStateDataType;

let getStateIsDebug = () =>
  CreateEditorStateDataEditorService.editorStateData.isDebug;

let setStateIsDebug = isDebug => {
  CreateEditorStateDataEditorService.editorStateData.isDebug = isDebug;

  ();
};

let getIsRun = () => CreateEditorStateDataEditorService.editorStateData.isRun;

let setIsRun = isRun => {
  CreateEditorStateDataEditorService.editorStateData.isRun = isRun;

  ();
};

let getState = () =>
  CreateEditorStateDataEditorService.editorStateData.editorState;

let setState = state => {
  CreateEditorStateDataEditorService.editorStateData.editorState = state;
  state;
};