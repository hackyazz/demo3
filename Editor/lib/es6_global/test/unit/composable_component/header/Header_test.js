

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as HeaderTool$WonderEditor from "./tool/HeaderTool.js";
import * as SettingTool$WonderEditor from "../../tool/SettingTool.js";
import * as DetectOSTool$WonderEditor from "./tool/DetectOSTool.js";
import * as ReactTestTool$WonderEditor from "../../../tool/ReactTestTool.js";
import * as BuildComponentTool$WonderEditor from "../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../tool/MainEditorSceneTool.js";

describe("Header", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return MainEditorSceneTool$WonderEditor.prepareScene(sandbox);
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test File", (function () {
                describe("test Controls", (function () {
                        describe("test modal content", (function () {
                                var _test = function (setOSFunc) {
                                  Curry._1(setOSFunc, /* () */0);
                                  StateEditorService$WonderEditor.setState(SettingTool$WonderEditor.setSetting(undefined, undefined, undefined, /* array */[
                                            /* record */[
                                              /* name */"focus",
                                              /* values : array */["f"]
                                            ],
                                            /* record */[
                                              /* name */"undo",
                                              /* values : array */[
                                                "ctrl+z",
                                                "command+z"
                                              ]
                                            ]
                                          ], /* () */0));
                                  return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildUI(HeaderTool$WonderEditor.buildFileComponent(HeaderTool$WonderEditor.buildState(undefined, undefined, true, undefined, undefined, undefined, undefined, /* () */0), Sinon.createEmptyStubWithJsObjSandbox(sandbox), undefined, undefined, /* () */0)));
                                };
                                Wonder_jest.test("if os is mac, should show command hot keys", (function () {
                                        return _test(DetectOSTool$WonderEditor.setOSToBeMac);
                                      }));
                                return Wonder_jest.test("else, shouldn't show command hot keys", (function () {
                                              return _test(DetectOSTool$WonderEditor.setOSToBeWin);
                                            }));
                              }));
                        return /* () */0;
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
