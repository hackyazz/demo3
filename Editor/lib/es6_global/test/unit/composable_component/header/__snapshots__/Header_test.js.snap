// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test File test Controls test modal content else, shouldn't show command hot keys 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title item-active"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      File
    </span>
  </div>
  <div
    className="item-content"
  >
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Undo
      </span>
    </div>
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Redo
      </span>
    </div>
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Controls
      </span>
    </div>
  </div>
  <article
    className="wonder-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Controls
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="content-field"
        >
          <div
            className="field-title"
          >
            focus
          </div>
          <div
            className="field-content"
          >
            f
          </div>
        </div>
        <div
          className="content-field"
        >
          <div
            className="field-title"
          >
            undo
          </div>
          <div
            className="field-content"
          >
            ctrl+z
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`Header test File test Controls test modal content if os is mac, should show command hot keys 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title item-active"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      File
    </span>
  </div>
  <div
    className="item-content"
  >
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Undo
      </span>
    </div>
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Redo
      </span>
    </div>
    <div
      className="content-section"
      onClick={[Function]}
    >
      <span
        className="section-header"
      >
        Controls
      </span>
    </div>
  </div>
  <article
    className="wonder-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Controls
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="content-field"
        >
          <div
            className="field-title"
          >
            focus
          </div>
          <div
            className="field-content"
          >
            f
          </div>
        </div>
        <div
          className="content-field"
        >
          <div
            className="field-title"
          >
            undo
          </div>
          <div
            className="field-content"
          >
            ctrl+z|command+z
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
